raven/base.py

```diff
@@ -636,7 +636,13 @@ def capture(self, event_type, data=None, date=None, time_spent=None,
             **kwargs)
 
         # should this event be sampled?
-        if self._random.random() < self.sample_rate:
+        sample_rate = self.sample_rate
+        try:
+            sample_rate = float(extra['sample_rate'])
```

Comment:
I would probably do a get here instead of exception and we'd want this to be an arg to capture rather than part of the extra metadata.